initfiles/examples/embed/couchbase-simple2.ecl

```diff
@@ -0,0 +1,140 @@
+            "batteryLevelPercentage": 95.32,
+            "bodyTemp": 30.08,
+            "coLevel": 1.46,
+            "forceSensitiveResistance": 166,
+            "heartRate": 94
+        },
+        "deviceTimestamp": "2017-06-05 15:04:35.924657-0400"
+    },
+    "receivedTimestamp": "1496275475926453"
+}
+*/
+
+deviceidrec := RECORD
+  string deviceID;
+END;
+
+rawdatarec := RECORD
+  real4 ambientTemp;
+  real4 barometer;
+  real4 batteryLevelPercentage;
+  real4 bodyTemp;
+  real4 coLevel;
+  real4 forceSensitiveResistance;
+  INTEGER heartRate;
+END;
+
+locationDatarec := RECORD
+  real x {xpath('x')};
+  real y {xpath('y')};
+  real z {xpath('z')};
+  string zoneId {xpath('zoneId')};
+END;
+
+messagerec := RECORD
+  string deviceTimestamp;
+  locationDatarec locationData;
+  rawdatarec rawdata;
+END;
+
+iotrec := RECORD
+  string deviceID;
+  string deviceType;
+  string receivedTimestamp;
+  messagerec message;
+END;
+
+BOOLEAN insertiotdoc(string docid, string doc) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1))
+  INSERT INTO iot (KEY, VALUE) VALUES ($docid, $doc) RETURNING TRUE AS c;
+ENDEMBED;
+
+BOOLEAN upsertiotdoc(string docid, string doc) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1))
+  INSERT INTO iot (KEY, VALUE) VALUES ($docid, $doc) RETURNING TRUE AS c;
+ENDEMBED;
+
+STRING getlatesttimestamp() := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1))
+  SELECT max(iot.receivedTimestamp) FROM iot;
+ENDEMBED;
+
+dataset(iotrec) getlatestdoc(string devid) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1))
+  SELECT iot.* FROM iot where iot.deviceID = $devid;
+ENDEMBED;
+
+integer devicecount() := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1))
+  SELECT count(DISTINCT iot.deviceID) FROM iot ;
+ENDEMBED;
+
+dataset(iotrec) devicebylocation(row(locationDatarec) values) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1), operation_timeout(5.5), config_total_timeout(15))
+  select iot.* from iot where iot.message.locationData.x = $x and iot.message.locationData.y = $y limit 1;
+ENDEMBED;
+
+dataset(deviceidrec) deviceidsbyzone(string zone) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1), operation_timeout(5.5), config_total_timeout(15))
+  select DISTINCT iot.deviceID from iot where iot.message.locationData.zoneId = $zone;
+ENDEMBED;
+
+sequential
+(
+  OUTPUT(insertiotdoc(adocid + random() + '"', adoc1));
+  Std.System.Debug.Sleep(200);
+  OUTPUT(insertiotdoc(adocid + random() + '"', adoc2));
+  OUTPUT(upsertiotdoc(adocid + random() + '"', adoc1));
+  OUTPUT(getlatesttimestamp());
+  Std.System.Debug.Sleep(200);
+  OUTPUT(getlatestdoc('"11135e49c-3c0f-446b-ad6a-4136deb88519"'));
+  OUTPUT(devicecount());
+  OUTPUT(devicebylocation(ROW({2.247,33.629,1.323,'"A"'},locationDatarec)));
+  OUTPUT(deviceidsbyzone('"A"'));
+  OUTPUT('Done');
+););
```

Comment:
The final ');' seems to be spurious.  Plus I think you should put a final newline at the end.